tests/components/history_stats/test_config_flow.py

```diff
@@ -193,3 +199,350 @@ async def test_entry_already_exist(
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+    assert result["preview"] == "history_stats"
+
+    with (
+        patch(
+            "homeassistant.components.recorder.history.state_changes_during_period",
+            _fake_states,
+        ),
+        freeze_time(t3),
+    ):
+        for end, exp_count in (
+            ("{{now()}}", "2"),
+            ("{{today_at('2:00')}}", "1"),
+            ("{{today_at('23:00')}}", "2"),
+        ):
+            await client.send_json_auto_id(
+                {
+                    "type": "history_stats/start_preview",
+                    "flow_id": result["flow_id"],
+                    "flow_type": "options_flow",
+                    "user_input": {
+                        CONF_ENTITY_ID: monitored_entity,
+                        CONF_TYPE: CONF_TYPE_COUNT,
+                        CONF_STATE: ["on"],
+                        CONF_END: end,
+                        CONF_START: "{{ today_at() }}",
+                    },
+                }
+            )
+
+            msg = await client.receive_json()
+            assert msg["success"]
+            assert msg["result"] is None
+
+            msg = await client.receive_json()
+            assert msg["event"]["state"] == exp_count
+
+    hass.states.async_set(monitored_entity, "on")
+
+    msg = await client.receive_json()
+    assert msg["event"]["state"] == "3"
+
+
+async def test_options_flow_preview_errors(
+    recorder_mock: Recorder,
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+) -> None:
+    """Test the options flow preview."""
+    logging.getLogger("sqlalchemy.engine").setLevel(logging.ERROR)
+    client = await hass_ws_client(hass)
+
+    # add state for the tests
+    monitored_entity = "binary_sensor.state"
+
+    # Setup the config entry
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={
+            CONF_NAME: DEFAULT_NAME,
+            CONF_ENTITY_ID: monitored_entity,
+            CONF_TYPE: CONF_TYPE_COUNT,
+            CONF_STATE: ["on"],
+            CONF_END: "{{ now() }}",
+            CONF_START: "{{ today_at() }}",
+        },
+        title=DEFAULT_NAME,
+    )
+    config_entry.add_to_hass(hass)
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    result = await hass.config_entries.options.async_init(config_entry.entry_id)
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+    assert result["preview"] == "history_stats"
+
+    for schema in (
+        {CONF_END: "{{ now() }"},
+        {CONF_START: "{{ today_at( }}"},
+        {CONF_DURATION: {"hours": 1}},
+    ):
```

Comment:
I don't fully understand what `schema` is doing here. It seems to overlap with `user_input` below. Probably I'm missing something

Reply:
This a for loop performing three separate tests of three separate invalid schemas. 

In the first loop, the first item in this tuple overwrites CONF_END in user input. 
In the second loop, the second item in this tuple overwrites CONF_START in user input.
In the third loop, the third item in this tuple overwrites CONF_DURATION in user input.

The other keys of USER_INPUT are common across all 3 loops.